### PR TITLE
Avoid recursion in `Interpreter::block`

### PR DESCRIPTION
**Pull request by abrown**
This resolves @bnjbvr's [comment](https://github.com/bytecodealliance/wasmtime/pull/1223#discussion_r400282578) in #1223.

### CODE REVIEW

File: cranelift/interpreter/src/interpreter.rs [hunk 1]
```diff
@@ -119,13 +119,16 @@ impl Interpreter {
...
         debug!("Block: {}", block);
-        for inst in frame.function.layout.block_insts(block) {
+        let layout = &frame.function.layout;
+        let mut maybe_inst = layout.first_inst(block);
+        while let Some(inst) = maybe_inst {
             match self.inst(frame, inst)? {
-                ControlFlow::Continue => continue,
+                ControlFlow::Continue => maybe_inst = layout.next_inst(inst),
                 ControlFlow::ContinueAt(block, old_names) => {
+                    debug!("Block: {}", block);
```
Comment (bnjbvr) [inline cranelift/interpreter/src/interpreter.rs:128]: Should we use a `trace!` here to be consistent with the other PR?

File: cranelift/interpreter/src/interpreter.rs [hunk 2]
```diff
@@ -119,13 +119,16 @@ impl Interpreter {
     /// Interpret a single [Block] in a [Function].
```
Comment (bnjbvr) [inline cranelift/interpreter/src/interpreter.rs:119]: pre-existing: this comment is lying just a bit, right? It interprets from a given block up to the function's return, if I'm not mistaken.

Comment (abrown) [reply cranelift/interpreter/src/interpreter.rs]: Good call... changed.

File: cranelift/interpreter/src/interpreter.rs (hunk 1 above)
Comment (abrown) [reply cranelift/interpreter/src/interpreter.rs]: Sure, added a commit for that.

**Comment by bnjbvr** [comment]
Thanks!

**Pull request merged**